source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -646,11 +607,9 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const string& name
                     inputSocket->setValue(bindParamValue);
 
                     input->setBindInput();
-                    ShaderNodePtr updateNode = graph->_nodeMap[newNodeName];
-                    graph->populateInputColorTransformMap(colorManagementSystem, updateNode, bindParam, targetColorSpace);
-                    ShaderInput* updateInput = updateNode->getInput(bindParam->getName());
```

Comment:
I don't think `input` is the correct one to send here. I had this before and got the wrong result as this is bindparam name, not the param name.

Reply:
I think the are the same input. The bindparam name and the param name must be the same here. We have ´BindParamPtr bindParam = shaderRef->getBindParam(elem->getName());` so the bindParam must be named the same as `elem->getName()`. And `input` is queried using `elem->getName()` so it must also be named the same. An the `updateNode`must be the same as `newNode` looking at how newNode was added to the graph above.

But let me know if I'm missing something.

Reply:
I wasn't sure but yes, it seems it's the case. Thanks for checking.